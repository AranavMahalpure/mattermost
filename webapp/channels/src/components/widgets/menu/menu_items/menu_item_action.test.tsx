// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {FormattedMessage} from 'react-intl';

import {renderWithContext, screen} from 'tests/react_testing_utils';

import MenuItemAction, {MenuItemActionImpl} from './menu_item_action';

jest.unmock('react-intl');

describe('components/MenuItemAction', () => {
    const baseProps = {
        onClick: jest.fn(),
    };

    test('should match snapshot', () => {
        const {container} = renderWithContext(
            <MenuItemActionImpl
                {...baseProps}
                text='Whatever'
            />,
        );

        expect(container).toMatchSnapshot();
    });
    test('should match snapshot with extra text', () => {
        const {container} = renderWithContext(
            <MenuItemActionImpl
                {...baseProps}
                text='Whatever'
                extraText='Extra Text'
            />,
        );

        expect(container).toMatchSnapshot();
    });

    test('should not translate the label and text when given strings', () => {
        renderWithContext(
            <MenuItemAction
                {...baseProps}
                ariaLabel='untranslated aria label'
                text='untranslated text'
            />,
        );

        expect(screen.getByLabelText('untranslated aria label')).toBeVisible();
        expect(screen.getByText('untranslated text')).toBeVisible();
    });

    test('should properly translate the label and text when given message descriptors', () => {
        renderWithContext(
            <MenuItemAction
                {...baseProps}
                ariaLabel={{id: 'MenuItemAction.ariaLabel'}}
                text={{id: 'MenuItemAction.text'}}
            />,
            {},
            {
                intlMessages: {
                    'MenuItemAction.ariaLabel': 'translated aria label',
                    'MenuItemAction.text': 'translated text',
                },
            },
        );

        expect(screen.getByLabelText('translated aria label')).toBeVisible();
        expect(screen.getByText('translated text')).toBeVisible();
    });

    test('should properly translate the text when given a FormattedMessage', () => {
        renderWithContext(
            <MenuItemAction
                {...baseProps}
                text={
                    <FormattedMessage id='MenuItemAction.message'/>
                }
            />,
            {},
            {
                intlMessages: {
                    'MenuItemAction.message': 'translated formatted message',
                },
            },
        );

        expect(screen.getByText('translated formatted message')).toBeVisible();
    });
});
