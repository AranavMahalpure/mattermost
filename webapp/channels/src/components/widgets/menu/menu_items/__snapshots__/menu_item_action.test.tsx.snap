// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MenuItemAction should match snapshot 1`] = `
<div>
  <button
    class="style--none"
  >
    <span
      class="MenuItem__primary-text"
    >
      Whatever
    </span>
  </button>
</div>
`;

exports[`components/MenuItemAction should match snapshot with extra text 1`] = `
<div>
  <button
    class="style--none MenuItem__with-help"
  >
    <span
      class="MenuItem__primary-text"
    >
      Whatever
    </span>
    <span
      class="MenuItem__help-text"
    >
      Extra Text
    </span>
  </button>
</div>
`;
